### DISCUSSION TITLE
Heritrix 3.4.0-SNAPSHOT-2022-03-08T19:15:59Z keeps pausing..

### DISCUSSION THREAD

**Discussion by cslr** [internetarchive/heritrix3#587] [Q&A] [answered]
Heritrx keeps pausing during crawls for no good reason. It may crawl something like 1GB and then pause again and again.

Only error message I see is "WARNING politessDelay unset" but I don't find documentation how to set politessDelay.

**Comment by anjackson** [comment, marked as answer]
The `politenessDelay` warning is irritating but should not be the cause of the pause.

As far as I know, pausing can happen for a couple of reasons:

1. You have a `DiskSpaceMonitor` configured ([e.g.](https://github.com/internetarchive/heritrix3/blob/d3ca691f8d111d9f9bd1f4ec9eecbbcf02942679/engine/src/main/resources/org/archive/crawler/restlet/profile-crawler-beans.cxml#L632)) and this is pausing the crawl because you don't have enough disk space given how that bean is configured.
2. You have less than 5GB of free space, and you are using the [default value of the FREE_DISK parameter for the underlying JE-BDB](https://docs.oracle.com/cd/E17277_02/html/java/com/sleepycat/je/EnvironmentConfig.html#FREE_DISK). See #340 for some information on that, but I personally would not recommend running Heritrix without plenty of space disk space.

If neither of these is consistent with what you're seeing then it sounds like a bug. In which case, some more detailed logs might help.
